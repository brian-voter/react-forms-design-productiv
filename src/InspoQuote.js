import React, { useState } from "react";
import axios from "axios";
import { v4 as uuid } from "uuid";

const QUOTE_API_URL = "https://inspo-quotes-api.herokuapp.com/quotes/random";

/** Component for inspirational quote.
 *
 * State:
 * - quote: object of {text, author}
 *
 * App -> InspoQuote -> { TodoForm, EditableTodoList }
 */

function InspoQuote() {
  const [quote, setQuote] = useState(null);

  /**
   * Updates the quote state object
   * @param {Object} newQuote
   */
  function updateQuote(newQuote) {
    setQuote(newQuote);
  }

  /**
   * Makes an API get request to fetch random quote
   * @returns Object - {text, author}
   */
  async function getQuote() {
    const result = await axios.get(QUOTE_API_URL);

    return result.data.quote;
  }

  /**
   * Handles click for when user clicks button
   */
  async function handleClick(evt) {
    updateQuote(await getQuote());
  }

  return (
    <div>
      {quote && <i id="quote-text">{quote.text} -{quote.author}</i>}
      <button id="quote-btn" onClick={handleClick}>
        {quote ? "Nü quøte" : "Click here for an inspirational quøte!"}
      </button>
    </div>
  )
}

export default InspoQuote;