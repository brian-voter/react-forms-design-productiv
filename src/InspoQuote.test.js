import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import InspoQuote from "./InspoQuote";

describe("Inspo Quote Component", function () {
  test("renders without crashing", function () {
    render(<InspoQuote />);
  });

  test("snapshot test", function () {
    const { container } = render(
      <InspoQuote />);
    expect(container).toMatchSnapshot();
  });

  test("rendered quotes app", function () {
    const result = render(<InspoQuote />);
    expect(result.queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();
  });
});

describe("able to display quotes", function () {
  test("first quote pops up", async function () {
    const { container, queryByText } = render(<InspoQuote />);
    expect(queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();

    const quoteBtn = container.querySelector("#quote-btn");
    fireEvent.click(quoteBtn);

    //must wait for the button click to finish api request
    await waitFor(() => {
      expect(queryByText('Nü quøte')).toBeInTheDocument();
    });
  });

  test("different quote pops up after second click", async function () {
    const { container, queryByText } = render(<InspoQuote />);
    expect(queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();

    //first button click
    const quoteBtn = container.querySelector("#quote-btn");
    fireEvent.click(quoteBtn);

    let firstQuote, secondQuote;
    await waitFor(() => {
      expect(queryByText('Nü quøte')).toBeInTheDocument();
      firstQuote = container.querySelector("#quote-text").textContent;
    });

    //second button click
    fireEvent.click(quoteBtn);

    await waitFor(() => {
      secondQuote = container.querySelector("#quote-text").textContent;
      expect(firstQuote).not.toEqual(secondQuote);
    });
  });
});