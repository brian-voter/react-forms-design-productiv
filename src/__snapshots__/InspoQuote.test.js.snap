// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inspo Quote Component snapshot test 1`] = `
<div>
  <div>
    <button
      id="quote-btn"
    >
      Click here for an inspirational qu√∏te!
    </button>
  </div>
</div>
`;
